### ISSUE TITLE
Getting all data from the profile collection working incorrectly

### ISSUE THREAD

**Issue by alekseykarpenko** [prescottprue/react-redux-firebase#914]
**Do you want to request a *feature* or report a *bug*?**  

**Bug**

**What is the current behavior?**

When I'm using `useFirestoreConnect` to get the data from collection, that was previously provided to store user profiles  (as `userProfile` option), I'm facing a very strange behavior, that seems like a bug. My component is being continuously updated, adding one new element to collection state after each iteration, while `isLoaded` is `true` from the very beginning.

Here is the code: 
```ts
export const Authors: React.FC<Props> = (props) => {
  useFirestoreConnect(
    {
      collection: 'users',
      storeAs: "authors"
    }
  )

  const users = useSelector((state:AppState) => state.firestore.ordered.authors)

  console.log(users, isLoaded(users));

  return (
    <div>

    </div>
  );
};
```

And here is console output:
![image](https://user-images.githubusercontent.com/4648983/79603768-9e55fe00-80f5-11ea-86f5-f4e0a145365b.png)

**What is the expected behavior?**
If I change collection to any other, that is not related to user profile, I meet the expected behavior:

![image](https://user-images.githubusercontent.com/4648983/79603684-7bc3e500-80f5-11ea-8f86-a8b9fcb18859.png)

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**

```json
{
    "firebase": "^7.13.1",
    "react": "^16.13.1",
    "react-redux": "^7.2.0",
    "react-redux-firebase": "^3.3.0",
    "redux": "^4.0.5",
    "redux-firestore": "^0.13.0",
}
```

**Comment by prescottprue** [comment]
Make sure you are passing an array of query configs to `useFirestoreConnect` instead of a single object

**Comment by alekseykarpenko** [comment]
@prescottprue Thanks for your quick response! I've changed the parameter, but the behavior left almost unchanged. Values are adding one by one.

**Comment by alekseykarpenko** [comment]
@prescottprue This strange behavior changes only if request another collection, that was not used as `userProfile`

**Comment by alekseykarpenko** [comment]
**UPD**: If I remove this from config, the same code works as expected, so it's definitely a bug related to `profile` and `useFirestoreConnect`:
```ts
const rrfConfig = {
  userProfile: 'users',
  useFirestoreForProfile: true
}
```

**Comment by alekseykarpenko** [comment]
UPD 2: I see a bunch of [DOCUMENT_ADDED](https://github.com/prescottprue/redux-firestore/blob/06e4bb47acac5246aac8145a4dd9d2cdd70454ad/src/constants.js#L78) actions in my Redux console., so it's thinking that i'm adding those documents into my collection at the same while i'm actually requesting them:
![image](https://user-images.githubusercontent.com/4648983/79615567-cbada680-810b-11ea-9662-f4a7922cfae1.png)

 @prescottprue Should this bug be related to `redux-firestore`?

**Comment by alekseykarpenko** [comment]
@prescottprue Could you please take a look on this bug? This is kind of blocker for me, as I was trying to implement some internal functionality with user profiles.

**Comment by alekseykarpenko** [comment]
@prescottprue Any thoughts on this?

**Comment by NiliusJulius** [comment]
I see the same thing happening. At the moment you cannot use user profiles and retrieve the users collection without this issue occurring.

**Comment by alekseykarpenko** [comment]
@prescottprue I have founded, that this bug is not related to `profiles` only, but to any situation, when you first connect the particular doc, and then query all the collection:
```
useFirestoreConnect([
  {
    collection: 'series',
    where: [
      ['default', '==', true],
      ['active', '==', true]
    ],
    limit: 1,
    storeAs: "defaultSeries"
  },
  {
    collection: 'series',
    where: [
      ['organizer.id', '==', organizerId]
    ],
    storeAs: "series-by-organizer"
  }
]
```

the second query in that case will trigger multiple `DOCUMENT_ADDED` events and rerender the Component after each update. This seems to me as very serious bug, you definitely should look on it.

**Comment by rgbutov** [comment]
I found the same wrong behavioral.
When I'm trying to get list of docs in child component if the parent doc get docs from the same collection i see multiple "@@reduxFirestore/DOCUMENT_ADDED" actions.

**Comment by prescottprue** [comment]
@alekseykarpenko Agreed that it is a big issue - It seems to be because of how Firebase's SDK is returning that data if it already has been loaded by the SDK in another listener.

[As seen here](https://github.com/prescottprue/redux-firestore/blob/d7a561361fecff128b35cf37712f852637a66f41/src/utils/query.js#L765) `docChanges` is being used to dispatch changes to documents if they exist and the updates size is less than the full query size. This was originally to handle updates which are triggered on only a single document or sub-group of documents instead of the whole list of docs.

There is a TODO there to make this optional, but wondering if that would cause any issues when only a single doc is changed.

**Comment by rajivsubra1981** [comment]
@alekseykarpenko and others - Thanks for catching this, it does hurt first-load performance. Any workarounds?

**Comment by rusakovic** [comment]
Confirm this issue.
Also `isLoaded` is not working correctly. Always shows `true`, if you reapply `where` query for `useFirestoreConnect`
 "react-redux-firebase": "^3.7.0",

**Comment by barthicus** [comment]
I also faced this problem. First load almost freezes the browser. Is there any temporary fix/hack? 🥺

**Comment by mike-lvov** [comment]
Hopefully, https://github.com/prescottprue/redux-firestore/pull/319 will fix this, when merged in

**Comment by potier97** [comment]
@mike-lvov, cuándo van a fusionar la rama para solucionar el problema? gracias

**Comment by 0xdevalias** [comment]
@potier97 I believe @mike-lvov is just an external person who wrote a PR to fix this issue. The timing of if/when this gets accepted/merged into the project/released is entirely up to @prescottprue (or another direct 'contributor' to this project, if there are any)

**Comment by mike-lvov** [comment]
@potier97 one way to handle it is to use my branch as a dependency in your package.json, but all the risks of doing it this way are on you.

**Comment by TrySpace** [comment]
For me this issue only arises when I come from a compontent/route that has queried a document `id` on the same collection, e.g. `postsById`, and then go to a list that queries `postsById`, it will trigger all the dreaded `DOCUMENT_ADDED`  like the console output in the original question.

It seems it's trying to be clever and 'smoothly' add the newly added documents, but I don't see any case where this will not impact performance. It happens generally (depending on latency) in the first few ms, so any transition animation will happen then, and gets interrupted. Setting a delay for render also does not help, because there is no control over when these documents are added.

But this does look promising: https://github.com/prescottprue/redux-firestore/pull/319

**Comment by prescottprue** [comment]
Would love to getting the mentioned PR merged, but [the tests are currently failing](https://github.com/prescottprue/redux-firestore/runs/1729316753#step:8:46) making me think that this may end up being a breaking change. Currently extremely busy, but if someone has a chance to look at the tests please reach out with your findings. I'll do the same when I get the chance to look at it
